x/go/binary/codec_test.go

```diff
@@ -75,4 +76,28 @@ var _ = Describe("Codec", func() {
 			Entry("MsgPack", &binary.MsgPackCodec{}),
 		)
 	})
+	Describe("Fallback", func() {
+		It("Should fallback to the next codec when the first one fails", func() {
+			js := &binary.JSONCodec{}
+			gb := &binary.GobCodec{}
+			type abc struct {
+				Value int `json:"value"`
+			}
+			v := abc{Value: 12}
+			jsonB := MustSucceed(js.Encode(nil, v))
+			gobB := MustSucceed(gb.Encode(nil, v))
+			var res abc
+			fbc := &binary.FallbackCodec{
```

Comment:
would have a test case for the situation i ask about above

Reply:
Sure